### PR TITLE
Validate collateral balance on loan generation

### PR DESCRIPTION
**Pull request by jmul-dev**
This PR will prevent user from submitting loan generation request unless they have sufficient balance and allowance for collateralization

### CODE REVIEW

File: src/modules/RequestLoan/RequestLoanForm/validator.ts [hunk 1]
```diff
@@ -27,3 +30,27 @@ export const validateInterestRate = (interestRate: number) => {
...
+
+export const validateCollateral = (tokens: TokenEntity[], collateral: any) => {
+    let selectedToken;
+    let response = { fieldName: "", error: "" };
+    for (let token of tokens) {
+        if (token.tokenSymbol === collateral.collateralTokenSymbol) {
+            selectedToken = token;
+            break;
+        }
+    }
```
Comment (chrismin) [inline src/modules/RequestLoan/RequestLoanForm/validator.ts]: Can we clean up the above for-loop with `find`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: src/modules/RequestLoan/RequestLoanForm/validator.ts [hunk 2]
```diff
@@ -27,3 +30,27 @@ export const validateInterestRate = (interestRate: number) => {
...
+    if (!selectedToken) {
+        response = {
+            fieldName: "collateralTokenSymbol",
+            error: `${collateral.collateralTokenSymbol} is currently not supported`,
+        };
+    } else if (
+        selectedToken &&
```
Comment (chrismin) [inline src/modules/RequestLoan/RequestLoanForm/validator.ts]: Unnecessary check (we already know `selectedToken` will return truthy).

**Comment by jmul-dev** [comment]
@chrismin fixed!

**Comment by chrismin** [comment]
lgtm!

**Pull request merged**